GitCommands/Git/Executable.cs

```diff
@@ -33,7 +33,8 @@ public Executable(Func<string> fileNameProvider, string workingDir = "")
                               bool redirectInput = false,
                               bool redirectOutput = false,
                               Encoding? outputEncoding = null,
-                              bool useShellExecute = false)
+                              bool useShellExecute = false,
+                              bool throwOnErrorOutput = true)
```

Comment:
How many callsites with `throwOnErrorOutput = true` vs `throwOnErrorOutput = false` do we have?

Reply:
42 (and 18 tests) that do not throw. Some of those could probably be tightened to not return errors.
Other about 69 (and 22 plugin and 24 tests)

Reply:
This is what this PR is about: Let the user see the errors *by default*.
I want to have a justification for all exceptions from this default.